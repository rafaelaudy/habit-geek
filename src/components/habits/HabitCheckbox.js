import React, { Fragment } from "react";

import "./HabitCheckbox.scss";

const HabitCheckbox = ({ isChecked, clickHandler, isDisabled, isReadOnly }) => {
  return (
    <Fragment>
      <label className="toggle">
        <input
          className="toggle__input"
          onChange={clickHandler}
          type="checkbox"
          checked={isChecked}
          disabled={isReadOnly ? false : isDisabled}
          data-readonly={isReadOnly}
        />
        <span className="toggle__label" data-readonly={isReadOnly}>
          <span className="toggle__text"></span>
        </span>
      </label>
    </Fragment>
  );
};

export default HabitCheckbox;
