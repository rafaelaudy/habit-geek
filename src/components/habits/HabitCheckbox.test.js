import React from "react";
import { shallow } from "enzyme";
import HabitCheckbox from "./HabitCheckbox";

describe("HabitCheckbox component", () => {
  it("renders static elements", () => {
    const habitCheckbox = shallow(<HabitCheckbox />);
    expect(habitCheckbox).toMatchSnapshot();
  });

  it("renders checked checkboxes", () => {
    const habitCheckbox = shallow(<HabitCheckbox isChecked={true} />);
    expect(habitCheckbox).toMatchSnapshot();
  });

  it("toggles the checkbox", () => {
    const clickHandlerMock = jest.fn();
    const dashboard = shallow(
      <HabitCheckbox clickHandler={clickHandlerMock} />
    );
    dashboard.find("input").simulate("change");
    expect(clickHandlerMock).toHaveBeenCalledTimes(1);
  });

  it("disables component", () => {
    const habitCheckbox = shallow(
      <HabitCheckbox isChecked={true} isDisabled={true} />
    );
    expect(habitCheckbox).toMatchSnapshot();
  });

  it("only disables component if it is not readonly", () => {
    const habitCheckbox = shallow(
      <HabitCheckbox isChecked={true} isDisabled={true} isReadOnly={true} />
    );
    expect(habitCheckbox).toMatchSnapshot();
  });
});
