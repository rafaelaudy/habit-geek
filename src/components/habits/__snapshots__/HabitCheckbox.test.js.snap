// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitCheckbox component disables component 1`] = `
<Fragment>
  <label
    className="toggle"
  >
    <input
      checked={true}
      className="toggle__input"
      disabled={true}
      type="checkbox"
    />
    <span
      className="toggle__label"
    >
      <span
        className="toggle__text"
      />
    </span>
  </label>
</Fragment>
`;

exports[`HabitCheckbox component only disables component if it is not readonly 1`] = `
<Fragment>
  <label
    className="toggle"
  >
    <input
      checked={true}
      className="toggle__input"
      data-readonly={true}
      disabled={false}
      type="checkbox"
    />
    <span
      className="toggle__label"
      data-readonly={true}
    >
      <span
        className="toggle__text"
      />
    </span>
  </label>
</Fragment>
`;

exports[`HabitCheckbox component renders checked checkboxes 1`] = `
<Fragment>
  <label
    className="toggle"
  >
    <input
      checked={true}
      className="toggle__input"
      type="checkbox"
    />
    <span
      className="toggle__label"
    >
      <span
        className="toggle__text"
      />
    </span>
  </label>
</Fragment>
`;

exports[`HabitCheckbox component renders static elements 1`] = `
<Fragment>
  <label
    className="toggle"
  >
    <input
      className="toggle__input"
      type="checkbox"
    />
    <span
      className="toggle__label"
    >
      <span
        className="toggle__text"
      />
    </span>
  </label>
</Fragment>
`;
